### PR TITLE
Document pubsub without cloudevent.

### LINKED ISSUE #1474: [CONTENT] Document pubsub without CE for 1.2
**What content needs to be created or modified?**
Document how pubsub can be used without cloud event, using the new `rawPayload` metadata.

**Describe the solution you'd like**
Modify the reference doc for pubsub mentioning the new metadata.

**Where should the new material be placed?**
https://docs.dapr.io/developing-applications/building-blocks/pubsub/
https://docs.dapr.io/developing-applications/building-blocks/pubsub/pubsub-overview/
https://docs.dapr.io/reference/api/pubsub_api/#metadata

**Additional context**

### PR DESCRIPTION
**Pull request by artursouza**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Document pubsub without cloudevent.

## Issue reference

This PR will close: #1474

### CODE REVIEW

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-overview.md [hunk 1]
```diff
@@ -100,6 +100,9 @@ Dapr can set a timeout message on a per message basis, meaning that if the messa
 
 - Note: Message TTL can also be set for a given queue at the time of component creation. Look at the specific characteristic of the component that you are using.
 
+### Communication with applications not using Dapr
+For scenarios where one application uses Dapr but another one doesn't, CloudEvent wrapping can be disabled as a publisher or subscriber. This feature is available since version 1.2 and allows partially adoption of Dapr pubsub in larger organizations. For more information, learn [how to use pubsub without CloudEvent]({{< ref pubsub-raw.md >}}).
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-overview.md]:
```suggestion
For scenarios where one application uses Dapr but another doesn't, CloudEvent wrapping can be disabled for a publisher or subscriber. This allows partial adoption of Dapr pubsub in large applications that cannot adopt Dapr all at once. For more information see [how to use pubsub without CloudEvent]({{< ref pubsub-raw.md >}}).
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-overview.md [hunk 2]
```diff
@@ -100,6 +100,9 @@ Dapr can set a timeout message on a per message basis, meaning that if the messa
 
 - Note: Message TTL can also be set for a given queue at the time of component creation. Look at the specific characteristic of the component that you are using.
 
+### Communication with applications not using Dapr
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-overview.md]:
```suggestion
### Communication with applications not using Dapr and CloudEvents
```

File: daprdocs/content/en/reference/api/pubsub_api.md [hunk 3]
```diff
@@ -60,6 +60,7 @@ Metadata can be sent via query parameters in the request's URL. It must be prefi
 Parameter | Description
 --------- | -----------
 metadata.ttlInSeconds | the number of seconds for the message to expire as [described here]({{< ref pubsub-message-ttl.md >}})
+metadata.rawPayload | publishes the event without wrapping it as CloudEvent as [described here]({{< ref pubsub-raw.md >}})
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/api/pubsub_api.md]:
```suggestion
metadata.rawPayload | boolean to determine if Dapr should publish the event without wrapping it as CloudEvent as [described here]({{< ref pubsub-raw.md >}})
```

File: daprdocs/content/en/reference/api/pubsub_api.md [hunk 4]
```diff
@@ -92,13 +93,24 @@ Example:
...
 
 > Note, all subscription parameters are case-sensitive.
 
+#### Metadata
+
+Optionally, metadata can be sent via the request body.
+
+Parameter | Description
+--------- | -----------
+rawPayload | subscribe to events that do not comply with CloudEvent specification, as [described here]({{< ref pubsub-raw.md >}})
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/api/pubsub_api.md]:
```suggestion
rawPayload | boolean to subscribe to events that do not comply with CloudEvent specification, as [described here]({{< ref pubsub-raw.md >}})
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
+---
+type: docs
+title: "Pub/Sub without CloudEvents"
+linkTitle: "Pub/Sub without CloudEvents"
+weight: 6000
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
weight: 7000
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+## Introduction
+
+Dapr uses CloudEvents to provide additional context to the event payload, enabling features like:
+* Tracing
+* Deduplication by message Id
+* Content-type for proper deserialization of event's data
+
+For more information about CloudEvent, please check the [official specification](https://github.com/cloudevents/spec).
+
+When adopting Dapr to an application, it might happen that other publisher or subscriber apps do not use Dapr and still require to communicate via Pub/Sub. For this reason, since version 1.2, Dapr enables apps to publish and subscribe to events that are not wrapped in a CloudEvent.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
When adding Dapr to your application, some services may still need to communicate via raw pub/sub messages not encapsulated in CloudEvents. This may be for compatibility reasons, or because some apps are not using Dapr yet. Dapr enables apps to publish and subscribe to raw events that are not wrapped in a CloudEvent.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+> Not using CloudEvent automatically disables support for tracing, deduplication per message Id, content-type metadata and any other feature built on top of the CloudEvent schema.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
{{% alert title="Warning" color="warning" %}}
Not using CloudEvents disables support for tracing, event deduplication per messageId, content-type metadata, and any other features built on top of the CloudEvent schema.
{{% /alert %}}
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+## Publish example
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
## Publishing raw messages
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+To disable CloudEvent wrapping, set the `rawPayload` metadata as part of the publishing request. This will enable subscribers to receive these messages without having to parsing the CloudEvent schema.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
To disable CloudEvent wrapping, set the `rawPayload` metadata to `true` as part of the publishing request. This will allow subscribers to receive these messages without having to parse the CloudEvent schema.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    $publisher = $factory->make(\Dapr\PubSub\Publish::class, ['pubsub' => 'pubsub']);
+    $publisher->topic('TOPIC_A')->publish('data', ['rawPayload' => 'true']);
+});
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+# Subscribe example
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
## Subscribing to raw events
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+When subscribing, add the additional metadata entry for `rawPayload` and Dapr sidecar will automatically wrap the payloads into a CloudEvent that is compatible with current Dapr SDKs.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
When subscribing programmatically, add the additional metadata entry for `rawPayload` so the Dapr sidecar will automatically wrap the payloads into a CloudEvent that is compatible with current Dapr SDKs.
```

Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]: Is the behavior here that setting rawPayload=true will cause the sidecar to wrap the raw event back into a CLoudEvent? What happens if this isn't set, will the app get a raw event without a Cloud Event?

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+import sys
+
+app = flask.Flask(__name__)
+CORS(app)
+
+@app.route('/dapr/subscribe', methods=['GET'])
+def subscribe():
+    subscriptions = [{'pubsubname': 'pubsub',
+                      'topic': 'deathStarStatus',
+                      'route': 'dsstatus',
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:109]: Please add a dsstatus route methid in the code to show the use of the Dapr SDK. Currently there's no SDK in this example

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md (hunk 11 above)
Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]: If this is not set, Dapr runtime will fail to parse the message and it will not work.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md (hunk 12 above)
Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:109]: Done.

**Comment by artursouza** [comment]
image

File: daprdocs/content/en/reference/api/pubsub_api.md [hunk 13]
```diff
@@ -92,13 +93,24 @@ Example:
   {
     "pubsubname": "pubsub",
     "topic": "newOrder",
-    "route": "/orders"
+    "route": "/orders",
+    "metadata": {
+      "rawPayload": "false",
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/api/pubsub_api.md]: Is there any scenario where someone would explicitly set rawPayload to false? Wouldn't the primary behavior be to just omit the value?

Asking because I don't want to show an example of something that's not the primary/recommended behavior.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+
+Dapr uses CloudEvents to provide additional context to the event payload, enabling features like:
+* Tracing
+* Deduplication by message Id
+* Content-type for proper deserialization of event's data
+
+For more information about CloudEvent, please check the [official specification](https://github.com/cloudevents/spec).
+
+When adding Dapr to your application, some services may still need to communicate via raw pub/sub messages not encapsulated in CloudEvents. This may be for compatibility reasons, or because some apps are not using Dapr yet. Dapr enables apps to publish and subscribe to raw events that are not wrapped in a CloudEvent.
+
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:19]:
```suggestion

<img src="/images/pubsub_raw.png" alt="Diagram showing how to publish with Dapr when subscriber does not use Dapr or CloudEvent" width=1000>

```

Comment (AaronCrawfis) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:19]: One single image is better at the top vs. two images is subsections.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+{{% alert title="Warning" color="warning" %}}
+Not using CloudEvents disables support for tracing, event deduplication per messageId, content-type metadata, and any other features built on top of the CloudEvent schema.
+{{% /alert %}}
+
+## Publishing raw messages
+
+<img src="/images/pubsub_publish_raw.png" alt="Diagram showing how to publish with Dapr when subscriber does not use Dapr or CloudEvent" width=1000>
+
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:29]:
```suggestion
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+});
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+## Subscribing to raw messages
+
+<img src="/images/pubsub_subscribe_raw.png" alt="Diagram showing how to subscribe with Dapr when publisher does not use Dapr or CloudEvent" width=1000>
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:84]:
```suggestion
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
I'd switch to one image at the top showing the end to end flow vs. separate images that show small subsections. Broken up images don't add much clarification to the flow

RawPubSub

File: daprdocs/content/en/reference/api/pubsub_api.md (hunk 13 above)
Comment (msfussell) [reply daprdocs/content/en/reference/api/pubsub_api.md]: @artursouza  - Can you address the question from @AaronCrawfis ?

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md (hunk 14 above)
Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:19]: Actually, these are two different scenarios. Raw pubsub is not supposed to be used when both applications use Dapr.

File: daprdocs/content/en/reference/api/pubsub_api.md (hunk 13 above)
Comment (artursouza) [reply daprdocs/content/en/reference/api/pubsub_api.md]: I've added something, so users can see how metadata can be returned. Yes, `false` is the default. I've changed it to true - but that is not the recommended value since there are many disadvantages of setting it to false and the publisher must also override this.

**Comment by artursouza** [comment]
@msfussell @AaronCrawfis  Anything else is needed in this PR?

**Review by msfussell — APPROVED** [review]

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md [hunk 17]
```diff
@@ -153,7 +153,7 @@ import sys
 app = flask.Flask(__name__)
 CORS(app)
 
-@app.route('/dsstatus', methods=['POST'])
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]: Why is this changed? In the above declaration it uses the dsstatus route:

```yml
apiVersion: dapr.io/v1alpha1
kind: Subscription
metadata:
  name: myevent-subscription
spec:
  topic: deathStarStatus
  route: /dsstatus
  pubsubname: pubsub
scopes:
- app1
- app2
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+
+When adding Dapr to your application, some services may still need to communicate via raw pub/sub messages not encapsulated in CloudEvents. This may be for compatibility reasons, or because some apps are not using Dapr. Dapr enables apps to publish and subscribe to raw events that are not wrapped in a CloudEvent.
+
+{{% alert title="Warning" color="warning" %}}
+Not using CloudEvents disables support for tracing, event deduplication per messageId, content-type metadata, and any other features built using the CloudEvent schema.
+{{% /alert %}}
+
+## Publishing raw messages
+
+<img src="/images/pubsub_publish_raw.png" alt="Diagram showing how to publish with Dapr when subscriber does not use Dapr or CloudEvent" width=1000>
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:28]: publisher

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+For more information about CloudEvents, read the [CloudEvents specification](https://github.com/cloudevents/spec).
+
+When adding Dapr to your application, some services may still need to communicate via raw pub/sub messages not encapsulated in CloudEvents. This may be for compatibility reasons, or because some apps are not using Dapr. Dapr enables apps to publish and subscribe to raw events that are not wrapped in a CloudEvent.
+
+{{% alert title="Warning" color="warning" %}}
+Not using CloudEvents disables support for tracing, event deduplication per messageId, content-type metadata, and any other features built using the CloudEvent schema.
+{{% /alert %}}
+
+## Publishing raw messages
+
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:25]:
```suggestion

Dapr apps are able to publish raw events to pub/sub topics without CloudEvent encapsulation, for compatibility with non-Dapr apps.

```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+});
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+## Subscribing to raw messages
+
+<img src="/images/pubsub_subscribe_raw.png" alt="Diagram showing how to subscribe with Dapr when publisher does not use Dapr or CloudEvent" width=1000>
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:84]: subscriber

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 21]
```diff
@@ -0,0 +1,149 @@
...
+    $publisher->topic('TOPIC_A')->publish('data', ['rawPayload' => 'true']);
+});
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+## Subscribing to raw messages
+
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:81]:
```suggestion

Dapr apps are also able to subscribe to raw events coming from existing pub/sub topics that do not use CloudEvent encapsulation.

```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 22]
```diff
@@ -0,0 +1,149 @@
...
+<img src="/images/pubsub_subscribe_raw.png" alt="Diagram showing how to subscribe with Dapr when publisher does not use Dapr or CloudEvent" width=1000>
+
+When subscribing programmatically, add the additional metadata entry for `rawPayload` so the Dapr sidecar automatically wraps the payloads into a CloudEvent that is compatible with current Dapr SDKs.
+
+{{< tabs "Python SDK" "PHP SDK" >}}
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion
{{< tabs "Python" "PHP SDK" >}}
```

Doesn't use SDK

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+    $publisher = $factory->make(\Dapr\PubSub\Publish::class, ['pubsub' => 'pubsub']);
+    $publisher->topic('TOPIC_A')->publish('data', ['rawPayload' => 'true']);
+});
+```
+
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+## Subscribing to raw messages
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:80]: Is there a declarative option?

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md (hunk 17 above)
Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/howto-publish-subscribe.md]: This should not have changed. Fixinf it.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md (hunk 18 above)
Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:28]: Thanks for the nice image. Added it.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md (hunk 20 above)
Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:84]: Thanks for the nice image. Added it.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md (hunk 23 above)
Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:80]: I just figured it out that our Subscription CRD lacks the metadata attribute:

https://github.com/dapr/dapr/issues/3225

Comment (artursouza) [reply daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:80]: BTW, I've added a note in this doc mentioning that this feature cannot be used in the declarative way because of that.

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 24]
```diff
@@ -0,0 +1,154 @@
...
+
+Dapr apps are also able to subscribe to raw events coming from existing pub/sub topics that do not use CloudEvent encapsulation.
+
+<img src="/images/pubsub_subscribe_raw.png" alt="Diagram showing how to subscribe with Dapr when publisher does not use Dapr or CloudEvent" width=1000>
+
+When subscribing programmatically, add the additional metadata entry for `rawPayload` so the Dapr sidecar automatically wraps the payloads into a CloudEvent that is compatible with current Dapr SDKs.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md:89]:
```suggestion

### Programmatically subscribe to raw events 

When subscribing programmatically, add the additional metadata entry for `rawPayload` so the Dapr sidecar automatically wraps the payloads into a CloudEvent that is compatible with current Dapr SDKs.
```

File: daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md [hunk 25]
```diff
@@ -0,0 +1,154 @@
...
+    }
+);
+
+$app->start();
+```
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+You cannot use this feature when subscribing to a topic using the Subscription Custom Resources Definition (CRD) since it does not contain the metadata attribute.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-raw.md]:
```suggestion

## Declaratively subscribe to raw events

Subscription Custom Resources Definitions (CRDs) do not currently contain metadata attributes ([issue #3225](https://github.com/dapr/dapr/issues/3225)). At this time subscribing to raw events can only be done through programmatic subscriptions.
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
Two final small updates

**Review by AaronCrawfis — APPROVED** [review]
